server/src/test/java/org/opensearch/cluster/routing/allocation/BalanceConfigurationTests.java

```diff
@@ -494,5 +913,97 @@ public ShardAllocationDecision decideShardAllocation(ShardRouting shard, Routing
             }
         }
     }
+}
+
+/**
+ * Utility class to show shards distribution across nodes.
+ */
+class ShardAllocations {
```

Comment:
This looks identical to the test component in SegmentReplicationAllocationIT - both tests are in the same project so we should be able to extract to a component.

Reply:
Yes you are right, though there is minor difference in how `RoutingNodes` are build in IT and Unit tests. I think it is possible to refactor this into a single class. Let me try to update this.